### ISSUE TITLE
Add lints to encourage use of specialized assert macros

### ISSUE THREAD

**Issue by Reisz** [rust-lang/rust-clippy#13252]
### What it does

Detect `assert!(a == b)` and `assert!(a != b)` and suggest using `assert_eq!` or `assert_ne!` respectively.

Should probably apply to `debug_assert!` as well.

### Advantage

- The specialized asserts provide more info in their error messages

### Example

```rust
assert!(fib(6) == 8);
```

Could be written as:

```rust
assert_eq!(fib(6), 8);
```

**Comment by nyurik** [comment]
I wonder if we should reuse the [bool_assert_comparison](https://rust-lang.github.io/rust-clippy/master/index.html#/bool_assert_comparison) lint.

The description actually matches what is being proposed:
> This lint warns about boolean comparisons in assert-like macros.
